server/src/test/java/org/elasticsearch/index/mapper/RuntimeFieldSourceProviderOptimizationTests.java

```diff
@@ -77,12 +86,15 @@ public void testWithSourceProviderOptimization() throws IOException {
                 var termQuery = fieldType.termQuery(32, context);
                 assertThat(searcher.count(termQuery), equalTo(1));
 
-                // Test that runtime based block loader works as expected with the optimization:
-                var blockLoader = fieldType.blockLoader(blContext(context.lookup()));
-                var columnReader = blockLoader.columnAtATimeReader(leafReaderContext);
-                var block = (TestBlock) columnReader.read(TestBlock.factory(), TestBlock.docs(leafReaderContext), 0, false);
-                for (int i = 0; i < block.size(); i++) {
-                    assertThat(block.get(i), equalTo((long) i));
+                // Test that script runtime field data works as expected with the optimization:
```

Comment:
Switched to field data in this test, because with change long script block loader no longer relies on scripts and the optimization that this test tests.